src/blog/2024/04/using-webhook-with-node-red.md

```diff
@@ -0,0 +1,188 @@
+
+## How Webhook works 
+
+!["Image displaying how webhook works"](./images/using-webhook-with-node-red-how-webhook-works.png "Image displaying how webhook works"){data-zoomable}
+
+- Event Initiator: This refers to the event specified to trigger the WebHook. Whenever this event occurs, the WebHook will be activated.
+- Webhook Server: The webhook server, is responsible for managing webhook configurations and endpoints. It listens for the specified event. When the event is detected, the webhook server automatically sends an HTTP POST request containing relevant data to the designated third-party application or service.
+- Data Reception by Third-Party Application: The third-party application will receive the data sent via the WebHook to the designated URL or listener provided during registration.
+- Custom Action Execution: Upon receiving the POST request, specific actions can be performed.
+
+## API Vs Webhook
+
+It's common and understandable to get confused between APIs and webhooks, especially when you are learning about webhooks for the first time. However, comparing the two can help dispel these confusions.
+
+## API:
+
+- **Direction:**
+  - Typically involves client-to-server communication.
+- **Initiation:**
+  - Requests are initiated by the client.
+- **Request Method:**
+  - Usually employs HTTP methods like GET, POST, etc.
+- **Response:**
+  - Provides an immediate response upon request.
+- **Data Transfer:**
+  - Utilizes a pull model where the client fetches data.
+- **Polling:**
+  - Requires periodic polling for updates.
+- **Payload:**
+  - The client specifies the payload in the request.
+- **Error Handling:**
+  - Typically includes error codes and messages.
+
+## Webhook:
+
+- **Direction:**
+  - Typically involves server-to-server communication.
+- **Initiation:**
+  - Requests are initiated by the server.
+- **Request Method:**
+  - Typically uses the HTTP POST method.
+- **Response:**
+  - Does not provide an immediate response; asynchronous.
+- **Data Transfer:**
+  - Operates on a push model where the server pushes data to the client.
+- **Polling:**
+  - No need for polling; receives updates directly.
+- **Payload:**
+  - The server defines the payload in the outgoing request.
+- **Error Handling:**
+  - Errors are handled by retry mechanisms or manual intervention.
+
+### Example Scenario:
+
+For example, Consider a manufacturing facility that utilizes temperature sensors to monitor temperature levels in critical areas. When the temperature surpasses predefined thresholds, it triggers a series of actions for maintenance and monitoring.
+
+!["Diagram explaining how component works in"](./images/using-webhook-with-node-red-diagram.png "Role Based Access Control For Node-RED with FlowFuse"){data-zoomable}
```

Comment:
Can you update the alt tag? this is a copy paste error I assume?

Reply:
Yes, it's a copy-paste error. I always begin by copying our previous blog post structure to prevent mistakes. I forgot to update this alt tag; I'll make sure to update it